src/fmi4c_utils.c

```diff
@@ -13,9 +13,10 @@ const char* getFunctionName(const char* modelName, const char* functionName) {
         return functionName;    //!< Do not change function name if model name is empty
     }
     char* fullName = (char*)malloc(strlen(modelName)+strlen(functionName)+2);
-    strcpy(fullName, modelName);
-    strcat(fullName, "_");
-    strcat(fullName, functionName);
+    strncpy(fullName, modelName, strlen(modelName));
```

Comment:
same problem in this strncpy case, (gcc compiler warns about it).
`warning: ‘__builtin_strncpy’ specified bound depends on the length of the source argument [-Wstringop-overflow=]`
Should be the size of fullName -1 (I think -1 for the null terminator to fit)

Things like this really make you appreciate C++